files/routes/volunteer_janitor.py

```diff
@@ -0,0 +1,92 @@
+
+from files.__main__ import app
+from files.classes.comment import Comment
+from files.classes.flags import CommentFlag
+from files.classes.user import User
+from files.classes.volunteer_janitor import VolunteerJanitorRecord, VolunteerJanitorResult
+from files.routes.volunteer_common import VolunteerDuty
+from flask import g
+import pprint
+import random
+import sqlalchemy
+
+class VolunteerDutyJanitor(VolunteerDuty):
+
+    def __init__(self, choices):
+        self.choices = choices
+
+    def accept(self, v) -> None:
+        for item in self.choices:
+            record = VolunteerJanitorRecord()
+            record.user_id = v.id
+            record.comment_id = item
+            record.recorded_utc = sqlalchemy.func.now()
+            record.result = VolunteerJanitorResult.Pending
+            g.db.add(record)
+        
+        g.db.commit()
+
+    def embed_template(self) -> str:
+        return "volunteer_janitor.html"
+    
+    def comments(self) -> list[Comment]:
+        return g.db.query(Comment).where(Comment.id.in_(self.choices))
+
+
+def get_duty(u: User) -> VolunteerDutyJanitor:
+    if not app.config['VOLUNTEER_JANITOR_ENABLE']:
+        return None
+
+    # these could probably be combined into one query somehow
+
+    # find reported comments not made by the current user
+    reported_comments = g.db.query(Comment) \
+        .where(Comment.filter_state == 'reported') \
+        .where(Comment.author_id != u.id) \
```

Comment:
consider also filtering out `is_approved` (I think that's the column name) as clicking the approve button is a mop action